lib/index.js

```diff
@@ -35,13 +35,23 @@ function traverse(input, strict) {
  */
 
 function object(obj, strict) {
+  // 'each' utility uses obj.length to check whether the obj is array. To avoid incorrect classification, wrap call to 'each' with rename of obj.length
+  if (obj.constructor !== Array && obj.hasOwnProperty('length')) {
```

Comment:
This won't quite work because they could pass `arguments` in a.js that would fail this test. I think we need 3 checks here:
- `obj.length` exists
- `obj.length` is a number
- `(objlength - 1) in obj`

Reply:
This could also actually fail on a normal array if it was `postMessage`d across frames

Reply:
> It actually is pretty standard to filter out a set of properties for an operation and add them back later
interesting, but do they usually create a clone of the parameters though no and then modify that as needed? 

@tsholmes what does `(obj.length - 1) in obj` prove? 🤔

Reply:
<img width="547" alt="screen shot 2017-09-06 at 11 25 56 am" src="https://user-images.githubusercontent.com/14844924/30128079-304213e4-92f6-11e7-8dc6-64dd9f5df430.png">

Reply:
> interesting, but do they usually create a clone of the parameters though no and then modify that as needed?

If we're modifying it in-place anyways its better to not make a full copy. If it was just an iteration you would probably not want to modify the input

Reply:
@hankim813 going to merge if you're ok with all of this